os/pm/pm_timer/pm_timer_set.c

```diff
@@ -0,0 +1,293 @@
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ************************************************************************/
+
+/************************************************************************
+ * Included Files
+ ************************************************************************/
+
+#include <tinyara/pm/pm.h>
+#include <tinyara/clock.h>
+#include <tinyara/irq.h>
+
+#include "pm_timer.h"
+
+/************************************************************************
+ * Pre-processor Definitions
+ ************************************************************************/
+
+/************************************************************************
+ * Private Type Declarations
+ ************************************************************************/
+
+/************************************************************************
+ * Public Variables
+ ************************************************************************/
+
+/************************************************************************
+ * Private Functions
+ ************************************************************************/
+
+pm_wakeup_timer_t *get_pm_timer(int id)
+{
+        /* If we also use malloc to allocate memory for timers,
+         * We need a good way to return the required timer ??? 
+         * For now, we just return timer from the g_pmTimer_list. */
+
+        /* handle the case if it is a valid id */
+        if (id < 0 || id >= CONFIG_PM_MAX_WAKEUP_TIMER) {
+                return NULL;
+        }
+
+        /* We should not do any operation with FREE status timers unless created by app*/
+        if (g_pmTimer_list[id].status == FREE) {
+                return NULL;
+        }
+
+        return &g_pmTimer_list[id];
+}
+
+void remove_pm_timer(pm_wakeup_timer_t *timer)
+{
+        (void)sq_rem((FAR sq_entry_t *)timer, &g_pmTimer_activeList);
+        timer->status = INACTIVE;
```

Comment:
remove from timer queue, then I think FREE is proper state. INACTIVE usually use for  something like a'stopped but not removed' state(c.f paused)

Reply:
yeah, the function is suppose to stop the timer. Not free. 
the name was kept remove pm timer becasue it removes from the list and will be used again.